api/app/resources/endpoints.py

```diff
@@ -0,0 +1,252 @@
+    if not user:
+        return False
+    g.user = user
+    return True
+
+bucketlist_items_fields = {
+    'id': fields.Integer,
+    'title': fields.String,
+    'bucketlist': fields.Integer,
+    'date_created': fields.String,
+    'date_modified': fields.String,
+    'done': fields.Boolean,
+}
+
+bucketlist_fields = {
+    'id': fields.Integer,
+    'name': fields.String,
+    'items': fields.Nested(bucketlist_items_fields),
+    'created_by': fields.String,
+    'date_created': fields.String,
+    'date_modified': fields.String,
+}
+
+class Limit(object):
+    """
+    This class's attribute helps maintains the limit (content per page limit)
+    across separate client page requests.
+    """
+    limit = 20
+
+class Bucketlists(Resource):
+    """
+    Bucket List Collection [/bucketlists/]
+    """
+
+    def __init__(self):
+        """Instantiate route request parameters"""
+        self.parser = reqparse.RequestParser()
+        self.parser.add_argument('name', type=str,
+                                 help='Enter name of Bucketlist', location='json')
+        self.parser.add_argument('created_by', type=str,
+                                 help="Creator of the bucketlist",
+                                 location='json')
+        super(Bucketlists, self).__init__()
+
+    @auth.login_required
+    def get(self, page=1):
+        """GET endpoint"""
+        bucketlists = []
+
+        try:
+            # Get the limit specified by the client
+            limit = int(request.args.get('limit', 0))
+        except ValueError:
+            # If limit specified by client isn't number type, ignore
+            # that and default to 20
+            limit = 20
+
+        if limit:
+            # if limit is greater than maximum, default to 100
+            if limit > 100:
+                Limit.limit = 100
+            elif limit < 1:
+            # if limit is <= 0, default to 20
+                Limit.limit = 20
+            else:
+                Limit.limit = limit
+
+        current_user = User.verify_auth_token(request.headers.get('token'))
+
+        # the "search bucketlist by name" parameter
+        search = False
+        qs = request.args.get('q')
+        if qs:
+            search = True
+        try:
+            page = int(request.args.get('page', 1))
+        except ValueError:
+            page = 1
+
+        if search:
+            query = Bucketlist.query.filter_by(
+                name=qs)
+            if query.count() == 0:
+                return {"message": "No search results"}
+            else:
+                info = {"message": "Search results"}
+                return [info, marshal(query.first(), bucketlist_fields)]
+
+        bucketlists = Bucketlist.query.all()
+        pagination = Pagination(per_page=Limit.limit, page=page,
+                                total=len(bucketlists), search=search,
+                                record_name='bucketlists')
+        info = {"Pagination Details": {
```

Comment:
**Code Quality: Nitpick**

You may need to change the format for the returned data.